d2game/d2player/skillicon.go

```diff
@@ -41,11 +37,24 @@ func newSkillIcon(ui *d2ui.UIManager, baseSprite *d2ui.Sprite, skill *d2hero.Her
 		lvlLabel:   label,
 	}
 
+	res.logger = d2util.NewLogger()
+	res.logger.SetLevel(l)
+	res.logger.SetPrefix(logPrefix)
+
 	res.SetPosition(x, y)
 
 	return res
 }
 
+type skillIcon struct {
+	*d2ui.BaseWidget
+	lvlLabel *d2ui.Label
+	sprite   *d2ui.Sprite
+	skill    *d2hero.HeroSkill
+
+	logger *d2util.Logger
+}
```

Comment:
another widget that has it's own logger. maybe this is not such an issue if you assign to it an existing logger (like from the HUD or some menu)

Reply:
maybe, so could you give some example. what should I type instead of `logger *d2util.Logger`?